test/inductor/test_max_autotune.py

```diff
@@ -0,0 +1,132 @@
+            mat3 = AlgorithmSelectorCache.benchmark_example_value(buf3)
+            mat4 = AlgorithmSelectorCache.benchmark_example_value(buf4)
+    
+            out = AlgorithmSelectorCache.benchmark_example_value(layout)
+            # expected_out = (mat1 @ mat2) + (mat3 @ mat4)
+            expected_out = None
+    
+            choice = aten_mm_plus_mm.bind((buf1, buf2, buf3, buf4), layout)
+            # use a tensor since the mutation to a python list in a sub process
+            # is not synced back to the parent process
+            timings = torch.zeros(3, dtype=torch.float32)
+            child = mp.Process(target=benchmark_choice, args=(choice, (mat1, mat2, mat3, mat4), out, expected_out, timings))
+            child.start()
+            child.join()
+            self.assertEqual(0, child.exitcode)
+            print(f"timings is {timings}, out {out}, expected_out {expected_out}")
+
+    def test_benchmark_choice_fail_in_subproc(self):
+        gm = make_fx(lambda: torch.zeros(2, 3))()  # a dummy graph to construct the GraphLowering
+        graph = GraphLowering(gm)
+
+        # the graph handler is neede to create benchmark example value below
+        with V.set_graph_handler(graph):
+            buf1 = self._create_buffer("mat1", (2, 3))
+            buf2 = self._create_buffer("mat2", (3, 2))
+            buf3 = self._create_buffer("mat3", (2, 3))
+            buf4 = self._create_buffer("mat4", (3, 2))
+    
+            layout = FixedLayout(torch.device("cuda:0"), torch.float32, (2, 2))
+    
+            mat1 = AlgorithmSelectorCache.benchmark_example_value(buf1)
+            mat2 = AlgorithmSelectorCache.benchmark_example_value(buf2)
+            mat3 = AlgorithmSelectorCache.benchmark_example_value(buf3)
+            mat4 = AlgorithmSelectorCache.benchmark_example_value(buf4)
+    
+            out = AlgorithmSelectorCache.benchmark_example_value(layout)
+            expected_out = (mat1 @ mat2) + (mat3 @ mat4)
+    
+            choice = FailChoiceCaller("fail_choice_caller", [], None)
+
+            # use a tensor since python list is not synced back
+            timings = torch.zeros(3, dtype=torch.float32)
+            child = mp.Process(target=benchmark_choice, args=(choice, (mat1, mat2, mat3, mat4), out, expected_out, timings))
+            child.start()
+            child.join()
+            self.assertNotEqual(0, child.exitcode)
+
+    def test_max_autotune_mm_plus_mm(self):
+        """
+        This crash previously due to a triton issue: https://github.com/openai/triton/issues/1298 .
+        With autotuning in subprocess, we don't crash anymore.
+        """
+        m, n, k = 2048, 1536, 64
+
+        def mm_plus_mm(a, b, c, d):
+            return a @ b + c @ d
+
+        a = torch.randn(m, k).cuda()
+        b = torch.randn(k, n).cuda()
+        c = torch.randn(m, k).cuda()
+        d = torch.randn(k, n).cuda()
+
+        with override_configs(max_autotune=True, autotune_in_subproc=True):
```

Comment:
We usually use `unittest.mock.patch` for this, see in test_torchinductor.py

Reply:
Ah, got it. I'll change the call and remove this API then